XX-Response-Time-Windows-for-Witness-Rewarding.md

```diff
@@ -0,0 +1,188 @@
+absolute fastest as absolute speeds provides no marginal utility, Uplink does not have a particular time-window, 
+donwlink time windows is up to 2 seconds, Join process up to 6 seconds.
+
+This HIP proposes to restore random hotspot selection but adding a response time window to eliminate only 
+slow hotspots that fail to meet LoRaWAN-grade timing constraints and push helium hotspots to improve their reponse time
+over time reasonibly. 
+
+# Motivation
+
+The LoRaWAN network has some timing constraints to be considered, these ones are related to the JOIN mechanism 
+and ACK/Downlink mechanism. JOIN requires a full loop within 5 seconds, up to 6. ACK/Downlink requires a full loop in 1 
+second for RX1 window, up to 2 seconds for RX2 windows. Out of this time frame, the response will be ignored by the devices.  [Appendix](#packet-processing-and-lorawan-time-constraints) gives the consequence of these constraints on the expected hotspot response time.
+
+LoRaWAN is a question of seconds, not a question of microseconds, this is why creating a competition between hotspots and network connectivity at a millisecond scale is not achieving any network goal.
+
+Hotspots with highly valuable locations, such as the mountaintops, cell towers, and even rooftops 
+sometimes rely on higher latency connectivity (4G/5G, Home Plug, Satellites) which adds anywhere from 
+10ms to 100ms. These hotspots generally have a higher operating cost due to dedicated connectivity and hosting cost
+and are unfairly impacted by the current selection algorithm as they still operate within LoRaWAN timing specifications. 
+See [appendix](#highly-valuable-coverage) about highly valuable coverage.
+
+Hotspot out of the city centers will get a slower Internet response time from the one in the city center, 
+even with fast Internet access, due to some extra network hops or downgraded connectivity technology like xDSL.
+See [Appendix](#suburb-valuable-coverage) about suburb valuable coverage.
+
+Hotspot with the worst locations, indoor, in the city center, gets the best response time experience 
+with direct fiber connectivity.
+
+We have seen that depending on the hardware of the hotspot, the Witness processing time is largely dependent 
+on the packet signature as described in the [appendix](#ecc-signature-impact). As the signature is delegated to an hardware 
+ECC chip for most of the deployed hotspot, the processing time depends on the hardware solution soldered. 
+Even if firmware can be improved, the current solution disqualifies certain hardware whatever is the hotspot owner's efforts. 
+
+More generally speaking, it disqualifies lower-end hardware like light-hotspots based on micro-controllers as, by definition, their capacity to process the same thing than an hotspot based on CPU is lower. This is nonsense as these hardware have a 
+better fit for stability, long-life, energy saving, lower cost, and should be privileged long term.
```

Comment:
Is there any proof backing this up? Because current light hotspots in the field do not exhibit the behavior you describe.

Reply:
Evidence does not really need to be shown.MCU computation power is lower than CPU  computation power. Code on MCU may be better optimized someway ... but gateway-rs code is the same for all. Currently time is mostly spent on ECC signature witch is equivalent and hide the other difference. Let's be a bit scientific or prove me that there will never be a difference of computation power in favor of CPI based system compared to MCU based system.